File: SampleAnalysisSourceCode/ConsoleApp1_VB/ConsoleApp1/Class1.vb
Imports ClassLibrary1
Imports Xxx = ClassLibrary1.BaseClass
Imports Yyy = ClassLibrary1

' コメント、ドキュメントコメント、継承のチェック

''' <summary>
''' ConsoleApp1 の Class1 です。
''' </summary>
Public Class Class1
    Inherits BaseClass
    Implements Interface1

    ''' <summary>
    ''' Name です。
    ''' </summary>
    ''' <returns></returns>
    Public Overrides Property ClassLibrary1_Interface1_Name As String
        Get
            Return MyBase.ClassLibrary1_Interface1_Name
        End Get
        Set(value As String)
            MyBase.ClassLibrary1_Interface1_Name = value
        End Set
    End Property

End Class

' Class1A です。1
''' <summary>
''' Class1A です。2
''' </summary>
Public Class Class1A

End Class

''' <summary>
''' Class1B です。1
''' </summary>
Public Class Class1B
    Inherits Class1A

End Class

' Class1C です。
Public Class Class1C
    Inherits Xxx

End Class


Public Class Class1D
    Inherits Yyy.BaseClass

End Class

Public Class Class1E
    Inherits Global.ClassLibrary1.BaseClass

End Class

File: SampleAnalysisSourceCode/ConsoleApp2_CSharp/ConsoleApp2/Class1.cs
using ClassLibrary1;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Xxx = ClassLibrary1.BaseClass;
using Yyy = ClassLibrary1;

// コメント、ドキュメントコメント、継承のチェック

namespace ConsoleApp2
{
    /// <summary>
    /// ConsoleApp2 の Class1 です。
    /// </summary>
    internal class Class1 : BaseClass, Interface1
    {
        /// <summary>
        /// Name です。
        /// </summary>
        public override string ClassLibrary1_Interface1_Name { get; set; }
    }

    // Class1A です。1
    /// <summary>
    /// Class1A です。2
    /// </summary>
    internal class Class1A { }

    /// <summary>
    /// Class1B です。1
    /// </summary>
    // Class1B です。2
    internal class Class1B : Class1A { }

    // Class1C です。
    internal class Class1C : Xxx { }

    internal class Class1D : Yyy.BaseClass { }

    internal class Class1E : global::ClassLibrary1.BaseClass { }
}
